pkg/tests/ossm/prometheus_scoped_scraping_test.go

```diff
@@ -0,0 +1,134 @@
+package ossm
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"github.com/maistra/maistra-test-tool/pkg/util/check/common"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/require"
+	"github.com/maistra/maistra-test-tool/pkg/util/env"
+	"github.com/maistra/maistra-test-tool/pkg/util/hack"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/shell"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+)
+
+func TestOperatorCanUpdatePrometheusConfigMap(t *testing.T) {
+	test.NewTest(t).Groups(test.Full).Run(func(t test.TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+
+		t.Log("This test checks if the operator can update Prometheus ConfigMap when the SMMR is updated")
+
+		meshNamespace := env.GetDefaultMeshNamespace()
+		defaultSMMR := shell.Executef(t, "oc -n %s get smmr default -o jsonpath='{.metadata.annotations.kubectl\\.kubernetes\\.io/last-applied-configuration}'", meshNamespace)
+
+		t.Cleanup(func() {
+			oc.ApplyString(t, meshNamespace, defaultSMMR)
+		})
+
+		t.LogStepf("Delete current SMMR %s", defaultSMMR)
+		oc.DeleteFromString(t, meshNamespace, defaultSMMR)
+
+		getPrometheusConfigCmd := fmt.Sprintf("oc -n %s get configmap prometheus -o jsonpath='{.data.prometheus\\.yml}'", meshNamespace)
+
+		t.NewSubTest("when creating a SMMR").Run(func(t test.TestHelper) {
+			ns := generateNamespace()
+
+			t.Cleanup(func() {
+				oc.DeleteNamespace(t, ns)
+			})
+
+			t.LogStepf("Create namespace %s and add it into SMMR", ns)
+			oc.CreateNamespace(t, ns)
+			updateDefaultSMMRWithNamespace(t, ns)
+
+			t.LogStepf("Look for %s in prometheus ConfigMap", ns)
+			shell.Execute(t, getPrometheusConfigCmd, checkForNamespace(ns))
```

Comment:
shell.Execute returns string now, so you can check something in the configmap is needed because now is only testing that command does not fail?

Reply:
When writing it, I did confirm with the source [shell_helper](https://github.com/maistra/maistra-test-tool/blob/da46ddcec785ab546474e5c3a7bf5df0cc052a46/pkg/util/shell/shell_helper.go#L32-L34). The `shell.Execute` is still performing checks on passed in `CheckFunc`.